sdks/java/io/hadoop-input-format/src/test/java/org/apache/beam/sdk/io/hadoop/inputformat/HadoopInputFormatIOIT.java

```diff
@@ -137,27 +153,74 @@ public void readUsingHadoopInputFormat() {
         .apply("Generate sequence", GenerateSequence.from(0).to(numberOfRows))
         .apply("Produce db rows", ParDo.of(new DeterministicallyConstructTestRowFn()))
         .apply("Prevent fusion before writing", Reshuffle.viaRandomKey())
+        .apply("Record write time", ParDo.of(new TimeMonitor<>(NAMESPACE, "write_time")))
         .apply(
             "Write using JDBCIO",
             JdbcIO.<TestRow>write()
                 .withDataSourceConfiguration(JdbcIO.DataSourceConfiguration.create(dataSource))
                 .withStatement(String.format("insert into %s values(?, ?)", tableName))
                 .withPreparedStatementSetter(new PrepareStatementFromTestRow()));
 
-    writePipeline.run().waitUntilFinish();
+    PipelineResult writeResult = writePipeline.run();
+    writeResult.waitUntilFinish();
 
     PCollection<String> consolidatedHashcode =
         readPipeline
             .apply(
                 "Read using HadoopInputFormat",
                 HadoopInputFormatIO.<LongWritable, TestRowDBWritable>read()
                     .withConfiguration(hadoopConfiguration.get()))
+            .apply("Record read time", ParDo.of(new TimeMonitor<>(NAMESPACE, "read_time")))
```

Comment:
(similar to above comment)

Reply:
Other than that I agree that we have no better way of collecting the time metric for this IO. To collect the most precise metric for that we would need to collect start time metrics inside the IO (as there is no way to place the monitor before PBegin and after PDone). 

BTW (maybe an opportunity to do another PR for this): do you think we should extend IOs with such metrics collection?

Reply:
I think such extension would be really useful. Do you know what kind of changes would be necessary to do it? Who could help?

Reply:
For now, I created an issue for this: https://issues.apache.org/jira/browse/BEAM-6969